piff/knn_interp.py

```diff
@@ -0,0 +1,373 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: knn_interp
+"""
+
+from __future__ import print_function
+
+from .interp import Interp
+from .starfit import Star, StarFit
+from .stardata import StarData
+
+import numpy
+import fitsio
+
+
+class kNNInterp(Interp):
+    """
+    An interpolator that uses sklearn KNeighborsRegressor to interpolate a
+    single surface
+    """
+    def __init__(self, logger=None, **knr_kwargs):
+        """Create the kNN interpolator
+
+        :param attr_interp: A list of star attributes to interpolate from
+        :param attr_target: A list of star attributes to interpolate to
+        :param logger:      A logger object for logging debug info. [default: None]
+        :param knr_kwargs:  Arguments that will be passed to the regressor.
+        """
+
+        self.kwargs = {
+            'n_neighbors': 15,
+            'weights': 'uniform',
+            'algorithm': 'auto',
+            'p': 2,
+            }
+        self.kwargs.update(knr_kwargs)
+        self.knn = {}
+
+    def build(self, attr_interp, attr_target, logger=None):
+        from sklearn.neighbors import KNeighborsRegressor
+        self.attr_interp = numpy.array(attr_interp)
+        self.attr_target = numpy.array(attr_target)
+        for target in self.attr_target:
+            self.knn[target] = KNeighborsRegressor(**self.kwargs)
+
+    def _fit(self, X, y, logger=None):
+        """Update the Neighbors Regressor with data
+
+        :param X:   The locations for interpolating. (n_samples, n_features)
+        :param y:   The target values. (n_samples, n_targets)
```

Comment:
We can do better than `X` and `y` for these parameters.  Give them names that mean what they are.